x-pack/plugin/inference/qa/test-service-plugin/src/main/java/org/elasticsearch/xpack/inference/mock/TestInferenceServicePlugin.java

```diff
@@ -100,11 +93,12 @@ public TransportVersion getMinimalSupportedVersion() {
 
     public abstract static class TestInferenceServiceBase implements InferenceService {
 
+        @SuppressWarnings("unchecked")
         private static Map<String, Object> getTaskSettingsMap(Map<String, Object> settings) {
             Map<String, Object> taskSettingsMap;
             // task settings are optional
             if (settings.containsKey(ModelConfigurations.TASK_SETTINGS)) {
-                taskSettingsMap = removeFromMapOrThrowIfNull(settings, ModelConfigurations.TASK_SETTINGS);
+                taskSettingsMap = (Map<String, Object>) settings.remove(ModelConfigurations.TASK_SETTINGS);
```

Comment:
I had problems using the code from `MapParsingUtils` in this plugin. It would compile but when the plugin was run the class loader complained that it could not find the `MapParsingUtils` class. I think because the plugin uses a different class loader, I didn't need that code in the test hence these changes.